src/function/gds/K_Core_Decomposition.cpp

```diff
@@ -0,0 +1,324 @@
+#include "binder/binder.h"
+#include "common/types/types.h"
+#include "function/gds/gds_frontier.h"
+#include "function/gds/gds_function_collection.h"
+#include "function/gds/gds_object_manager.h"
+#include "function/gds/gds_utils.h"
+#include "function/gds/output_writer.h"
+#include "function/gds_function.h"
+#include "graph/graph.h"
+#include "processor/execution_context.h"
+#include "processor/result/factorized_table.h"
+
+using namespace kuzu::binder;
+using namespace kuzu::common;
+using namespace kuzu::processor;
+using namespace kuzu::storage;
+using namespace kuzu::graph;
+
+namespace kuzu {
+namespace function {
+
+class KUZU_API KCoreFrontierPair : public FrontierPair {
+public:
+    KCoreFrontierPair(std::shared_ptr<GDSFrontier> curFrontier,
+        std::shared_ptr<GDSFrontier> nextFrontier, uint64_t maxThreads,
+        table_id_map_t<offset_t> numNodesMap, storage::MemoryManager* mm)
+        : FrontierPair(curFrontier, nextFrontier, maxThreads), numNodesMap{numNodesMap} {
+        for (const auto& [tableID, curNumNodes] : numNodesMap) {
+            curVertexValues.allocate(tableID, curNumNodes, mm);
+            auto data = curVertexValues.getData(tableID);
+            for (auto i = 0u; i < curNumNodes; ++i) {
+                data[i].store(0, std::memory_order_relaxed);
+            }
+        }
+    }
+
+    void initRJFromSource(common::nodeID_t /* source */) override { setActiveNodesForNextIter(); };
```

Comment:
is this used? If not, why implement this function with `setActiveNodesForNextIter();`